marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/CallbackGeneratorTest.java

```diff
@@ -0,0 +1,415 @@
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.util.Date;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLStreamReader;
+import javax.xml.transform.Source;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.w3c.dom.Document;
+import org.xml.sax.InputSource;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.MarkLogicInternalException;
+import com.marklogic.client.dataservices.CallBatcher.CallArgsGenerator;
+import com.marklogic.client.dataservices.CallManager;
+import com.marklogic.client.dataservices.CallManager.CallArgs;
+import com.marklogic.client.dataservices.impl.CallBatcherImpl;
+import com.marklogic.client.dataservices.impl.CallBatcherImpl.BuilderImpl;
+import com.marklogic.client.dataservices.impl.CallManagerImpl.CallerImpl;
+import com.marklogic.client.io.marker.AbstractWriteHandle;
+import com.marklogic.client.test.Common;
+
+public class CallbackGeneratorTest implements CallArgsGenerator<CallManager.CallEvent>{
+
+    private static DatabaseClient db = Common.connect();
+    private static CallerImpl callerImpl;
+    private static CallbackGeneratorTest callbackGeneratorTest;
+    
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @BeforeClass
+    public static void setup() {
+        callerImpl = new CallerImpl<CallManager.CallEvent>(null) {
+
+            @Override
+            public CallManager.CallEvent callForEvent(DatabaseClient client, CallArgs args) throws Exception {
+                return null;
+            }
+        };
+        callbackGeneratorTest = new CallbackGeneratorTest();
+    }
+
+    @Test(expected = Test.None.class)
+    public void forArgsGeneratorTest() {
+        BuilderImpl builderImpl = new BuilderImpl(db,callerImpl);
+        assertNotNull(builderImpl.forArgsGenerator(callbackGeneratorTest));
+    }
+    
+    @Test
+    public void voidInputTypeTest() {
+        
+        CallBatcherImpl batcherImpl = new CallBatcherImpl(db, callerImpl, Void.class);
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage(startsWith("Cannot call add() when supplying arguments with generator."));
+        batcherImpl.add(null);
+    }
+    
+    @Test
+    public void callbatcherInitializeTest() {
+        CallBatcherImpl batcherImpl = new CallBatcherImpl(db, callerImpl, Void.class, callbackGeneratorTest);
+        expectedException.expect(MarkLogicInternalException.class);
+        expectedException.expectMessage(startsWith("Unsupported implementation of call arguments."));
+        batcherImpl.start(null);
+    }
+
+    @Override
+    public CallArgs apply(CallManager.CallEvent t) {
+        return new CallArgs() {
```

Comment:
This implementation of CallArgs is useful for callbatcherInitializeTest() but can't be used for any test of successful calls.